### DISCUSSION TITLE
I am new to react hook form, how can i achieve the field level validation for input field

### DISCUSSION THREAD

**Discussion by mohanagowthami** [react-hook-form/react-hook-form#2574] [Q&A]
when the form is submitting, it is validating all fields, but I want field-level validation like when the email format is wrong, when we moved to the below field , immediately we have to see the error. before submitting the form

**Comment by bluebill1049** [comment]
have a read these two sections: 

https://react-hook-form.com/api#useForm

https://react-hook-form.com/api#register

**Reply by mohanagowthami** [reply]
Thanks, I have to implement in **react native**, I am using the Controller

**Reply by bluebill1049** [reply]
have a look at the Controller section then, `rules`

**Reply by mohanagowthami** [reply]
here, How can I implement this through the Controller in react-native, is there any solution

**Reply by mohanagowthami** [reply]
thank u

**Reply by bluebill1049** [reply]
https://react-hook-form.com/get-started#ReactNative
